contrib/mesos/pkg/executor/executor.go

```diff
@@ -208,23 +207,81 @@ func (k *KubernetesExecutor) isDone() bool {
 	}
 }
 
+func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
+	k.lock.RLock()
+	defer k.lock.RUnlock()
+
+	ni = k.nodeInfo
+	return
+}
+
+func (k *KubernetesExecutor) onRegister(ei *mesos.ExecutorInfo, si *mesos.SlaveInfo) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	if !(&k.state).transition(disconnectedState, connectedState) {
+		log.Errorf("failed transitioning to a connected state")
+	}
+
+	// update labels of node
+	// TODO(sttts): enable when node-labels branch is merged
+	//if k.client != nil && si != nil {
+	//	node.CreateOrUpdate(k.client, si.GetHostname(), node.SlaveAttributesToLabels(si.GetAttributes()))
+	//}
+
+	// get executor resources
+	if ei != nil {
+		for _, r := range ei.GetResources() {
+			if r == nil || r.GetType() != mesos.Value_SCALAR {
+				continue
+			}
+			switch r.GetName() {
+			case "cpus":
+				k.executorCPU = r.GetScalar().GetValue()
+			case "mem":
+				k.executorMem = r.GetScalar().GetValue()
+			}
+		}
+	}
+
+	// get resource capacity of the node
+	for _, r := range si.GetResources() {
```

Comment:
do we need to check for role="something" here? probably not... but not 100% sure